lib/auth/init.go

```diff
@@ -483,6 +482,49 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 	return asrv, nil
 }
 
+func setAuthPreference(asrv *Server, newAuthPref services.AuthPreference) error {
```

Comment:
nit: `initSetAuthPreference` to prevent someone accidentally calling it from a non-init codepath